task/clamav-scan/0.1/clamav-scan.yaml

```diff
@@ -63,40 +65,101 @@ spec:
-          --alert-phishing-ssl=yes --alert-phishing-cloak=yes --alert-partition-intersection=yes \
-          | tee /tekton/home/clamscan-result.log || true
-        echo "Executed-on: Scan was executed on version - $(clamscan --version)" | tee -a /tekton/home/clamscan-result.log
-
-        # OPA/EC requires structured data input, add clamAV log into json
-        jq -Rs '{ output: . }' /tekton/home/clamscan-result.log > /tekton/home/clamscan-result-log.json
-
-        EC_EXPERIMENTAL=1 ec test \
-          --namespace required_checks \
-          --policy /project/clamav/virus-check.rego \
-          -o json \
-          /tekton/home/clamscan-result-log.json || true
-
-        # workaround: due to a bug in ec-cli, we cannot generate json and appstudio output at the same time, running it again
-        EC_EXPERIMENTAL=1 ec test \
-          --namespace required_checks \
-          --policy /project/clamav/virus-check.rego \
-          -o appstudio \
-          /tekton/home/clamscan-result-log.json | tee $(results.TEST_OUTPUT.path) || true
+        for file in /tekton/home/clamscan-result-*.log; do
+          if [ -e "$file" ]; then
+            file_suffix=$(basename "$file" | sed 's/clamscan-result-//;s/.log//')
+            # OPA/EC requires structured data input, add clamAV log into json
+            jq -Rs '{ output: . }' /tekton/home/clamscan-result-$file_suffix.log > /tekton/home/clamscan-result-log-$file_suffix.json
+
+            EC_EXPERIMENTAL=1 ec test \
+              --namespace required_checks \
+              --policy /project/clamav/virus-check.rego \
+              -o json \
+              /tekton/home/clamscan-result-log-$file_suffix.json || true
+
+            # workaround: due to a bug in ec-cli, we cannot generate json and appstudio output at the same time, running it again
+            EC_EXPERIMENTAL=1 ec test \
+              --namespace required_checks \
+              --policy /project/clamav/virus-check.rego \
+              -o appstudio \
+              /tekton/home/clamscan-result-log-$file_suffix.json | tee /tekton/home/clamscan-ec-test-$file_suffix.json || true
+
+            cat /tekton/home/clamscan-ec-test-$file_suffix.json
+          fi
+        done
+
+        jq -s -rce '
+          reduce .[] as $item ({"timestamp":"0","namespace":"","successes":0,"failures":0,"warnings":0,"result":"SUCCESS","note":"All checks passed successfully"};
+            {
+            "timestamp" : (if .timestamp < $item.timestamp then $item.timestamp else .timestamp end),
+            "namespace" : $item.namespace,
+            "successes" : (.successes + $item.successes),
+            "failures" : (.failures + $item.failures),
+            "warnings" : (.warnings + $item.warnings),
+            "result" : (if $item.result != "SUCCESS" then $item.result else .result end),
+            "note" : (if $item.result != "SUCCESS" then $item.note else .note end)
```

Comment:
This part of the result needs to be refined. Currently, it returns the `result` and `note` of the scan result that is not `SUCCESS` but as mentioned in https://github.com/redhat-appstudio/architecture/blob/main/ADR/0030-tekton-results-naming-convention.md#test_output-schema-validation, the result can be any of `["SUCCESS", "FAILURE", "WARNING", "SKIPPED", "ERROR"]`. So in an aggregated result, we probably need to rank the result statuses. I do not see that currently in the ADR.

As of today, the current approach may work for multi-arch images that build for two arches but the approach is not robust for builds supporting 3 or more arches. @arewm @ralphbean @dirgim @MartinBasti @lcarva Can I please request you to share your thoughts on this?

Reply:
Great catch!